packages/block-library/src/query/edit/block-setup/index.js

```diff
@@ -0,0 +1,101 @@
+/**
+ * WordPress dependencies
+ */
+import { useSelect } from '@wordpress/data';
+import { useBlockProps } from '@wordpress/block-editor';
+import { store as blocksStore } from '@wordpress/blocks';
+import { useState, useMemo } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+import { Placeholder, Button } from '@wordpress/components';
+
+/**
+ * Internal dependencies
+ */
+import LayoutSetupStep from './layout-step';
```

Comment:
It seems fine to start with the query block, but let's write the component with an eye towards making it part of the generic `<Placeholder>` whenever a block has registered patterns as per #28735.

Reply:
I had this a bit in mind with `BlockSetup`. If we can consolidate the use cases of more blocks I'll look into it better. It's mostly a prototype now.